relays/lib-substrate-relay/src/parachains/source.rs

```diff
@@ -22,35 +22,34 @@ use async_std::sync::{Arc, Mutex};
 use async_trait::async_trait;
 use bp_parachains::parachain_head_storage_key_at_source;
 use bp_polkadot_core::parachains::{ParaHash, ParaHead, ParaHeadsProof, ParaId};
+use bp_runtime::HeaderIdProvider;
 use codec::Decode;
 use parachains_relay::{
-	parachains_loop::{ParaHashAtSource, SourceClient},
-	parachains_loop_metrics::ParachainsLoopMetrics,
+	parachains_loop::SourceClient, parachains_loop_metrics::ParachainsLoopMetrics,
 };
 use relay_substrate_client::{
 	Chain, Client, Error as SubstrateError, HeaderIdOf, HeaderOf, RelayChain,
 };
-use relay_utils::relay_loop::Client as RelayClient;
-use sp_runtime::traits::Header as HeaderT;
+use relay_utils::{relay_loop::Client as RelayClient, NoopOption};
 
 /// Shared updatable reference to the maximal parachain header id that we want to sync from the
 /// source.
-pub type RequiredHeaderIdRef<C> = Arc<Mutex<Option<HeaderIdOf<C>>>>;
+pub type RequiredHeaderIdRef<C> = Arc<Mutex<NoopOption<HeaderIdOf<C>>>>;
```

Comment:
This actually seems like a redundant change - we'll be only using `Some` or `None` here. right? :) So we don't need this `Noop` at all. But I'm fine with leaving it as is :)